Assignments/SCF-Session-3/Marksheet/Main.java

```diff
@@ -0,0 +1,177 @@
+package Marksheet;
+
+import java.util.Scanner;
+
+/**
+ * Contains all the methods to calculate average, maximum &
+ * minimum grades & the percent of students passed.
+ *
+ * @author Ishita Mathur
+ */
+class Marksheet
+{
+	int totalStudents; //assume totalStudents > 0
+	float[] grade; //assume grade >= 0.00 && grade <= 100.00
+	
+	public int getTotalStudents()
+	{
+		return totalStudents;
+	}
+	
+	public void setTotalStudents(int totalStudents)
+	{
+		this.totalStudents = totalStudents;
+	}
+	
+	public float[] getGrade()
+	{
+		return grade;
+	}
+	
+	public void setGrade(float[] grade)
+	{
+		this.grade = grade;
+	}
+	
+	/**
+	 * Finds average grade of all students.
+	 * @return average that is average grade of all students
+	 */
+	public float avgGrade() throws ArithmeticException
+	{
+		int i;
+		float totalScore = 0, average;
+		for(i = 0; i < totalStudents; i++)
+		{
+			totalScore += grade[i];
+		}
+		average = totalScore / totalStudents;
+		return average;
+	}
+	
+	/**
+	 * Finds maximum grade from all grades of students.
+	 * @return max that is the maximum grade
+	 */
+	public float maxGrade() throws ArithmeticException
+	{
+		float max = 0;
+		int i;
+		for( i = 0; i < totalStudents; i++)
+		{
+			if(max < grade[i])
+			max = grade[i];
+		}
+		return max;
+	}
+	
+	/**
+	 * Finds minimum grade from all grades of students.
+	 * @return min that is the minimum grade
+	 */
+	public float minGrade() throws ArithmeticException
+	{
+		float min = grade[0];
+		for(int i = 0; i < totalStudents; i++)
+		{
+			if(min > grade[i])
+			min = grade[i];
+		}
+		return min;
+	}
+	
+	/**
+	 * Finds percentage of students that passed. Pass is grade>=40.
+	 * @return percentPass that is the percentage of student passed
+	 */
+	public float studentsPass() throws ArithmeticException 
+	{
+		float totalPass = 0;
+		float percentPass;
+		
+		for(int i = 0; i < totalStudents; i++)
+		{
+			if(grade[i] >= 40)
+			totalPass++;
+		}
+		
+		percentPass = (totalPass / totalStudents) * 100;
+		
+		return percentPass;
+	}
+}
+
+public class Main 
```

Comment:
Language keywords should not be used as user defined items. `Main`

Reply:
done